src/components/Assistant.vue

```diff
@@ -302,7 +305,9 @@ export default {
 			})
 		},
 		async insertResult(task) {
-			this.$editor.commands.insertContent(task.output.output)
+			const isMarkdown = hasMarkdownSyntax(task.output.output) && isValidMarkdown(task.output.output)
```

Comment:
I think we could even just have one function `shouldInterpretAsMarkdown` or similarly named that does both checks, but not a blocker from my side.